Semantics.md

```diff
@@ -678,6 +829,51 @@ outside the range which rounds to an integer in range) traps.
     This trap is intended to be impossible for user code to catch or handle, even in the future when it may be possible to
     handle some other kinds of traps or exceptions.
 
+## Thread operators
+
+  * `is_lock_free`: Given an operand `N`, if the atomic step of an atomic
+    primitive (see [Atomic Memory Accesses](#atomic-memory-accesses)) on a
+    datum of size `N` bytes will be performed without the agent acquiring a
+    lock outside the `N` bytes comprising the datum, then return `1`.
+    Otherwise, return `0`.
+
+### Wait
+
+The wake and wait operators are optimizations over busy-waiting for a value to
+change. It is a validation error to use these operators on non-shared linear
+memory.
+
+The wait operators take three operands: an address operand, an expected
+value, and a relative timeout in milliseconds as an f64. The timeout must be
+non-negative, and can also be positive infinity. If the timeout is positive
+infinity then the timeout will never expire.
```

Comment:
Why disallow negative values? A timer would just be expired if it were negative. Otherwise, specify the effects of a negative timer.

What of NaN?